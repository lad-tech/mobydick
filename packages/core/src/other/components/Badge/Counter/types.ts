import {StyleProp, TextStyle, ViewStyle} from 'react-native';

import {TypographyProp} from '../../../../typography';

export enum ICounterSize {
  medium = 'medium',
  small = 'small',
}

export enum ICounterTypes {
  accentLight = 'accentLight',
  accent = 'accent',
  mutedLight = 'mutedLight',
  attentionLight = 'attentionLight',

  attention = 'attention',
  muted = 'muted',
}

export interface ICounterProps {
  count?: number;
  font?: TypographyProp;
  type?: ICounterTypes;
  style?: StyleProp<ViewStyle>;
  textStyle?: StyleProp<TextStyle>;
  size?: ICounterSize;
  maxLength?: number;
}
