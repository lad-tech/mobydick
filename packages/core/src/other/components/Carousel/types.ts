import {FlatListProps} from 'react-native/Libraries/Lists/FlatList';
import {ViewStyle} from 'react-native';
import {ReactElement} from 'react';

export enum ICarouselAlign {
  center = 'center',
  start = 'start',
}

export interface ICarouselProps<T>
  extends Omit<FlatListProps<T>, 'renderItem' | 'keyExtractor'> {
  data: Array<T>;
  sliderItem: (item: T, index: number, data: Array<T>) => ReactElement;
  keyExtractor: (item: T, index?: number) => string;
  itemWidth: number;
  sideMargin?: number;
  loading?: boolean;
  onPressItem?: (item: T) => void;
  activeItemId?: string;
  isDots?: boolean;
  averageItemLength?: number;
  animateAutoScroll?: boolean;
  onActiveChange?: (item: T) => void;
  align?: ICarouselAlign;
  isScrolling?: boolean;
  ms?: number;
  indexScroll?: number;
  dotSize?: number;
  activeDotColor?: string;
  passiveDotColor?: string;
  dotsStyles?: ViewStyle | ViewStyle[];
  isLoop?: boolean;
}
