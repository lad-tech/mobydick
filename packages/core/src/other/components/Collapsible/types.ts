import {
  LayoutAnimationProperty,
  LayoutAnimationType,
  TextStyle,
  TouchableOpacityProps,
  ViewStyle,
} from 'react-native';
import {ReactElement} from 'react';

import {TypographyProp} from '../../../typography';

export interface ICollapsibleProps extends TouchableOpacityProps {
  title?: string;
  duration?: number;
  containerStyle?: ViewStyle | ViewStyle[];
  headerStyle?: ViewStyle | ViewStyle[];
  titleStyle?: TextStyle | TextStyle[];
  fontTitle?: TypographyProp;
  typeAnimation?: LayoutAnimationType;
  creationPropAnimation?: LayoutAnimationProperty;
  numberOfLines?: number;
  initialState?: boolean;
  isAnimated?: boolean;
  titleBottomView?: ReactElement;
}
