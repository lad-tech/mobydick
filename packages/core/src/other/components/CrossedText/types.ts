import {IStyledTextProps} from '../../../typography';

export interface ICrossedTextProps extends IStyledTextProps {
  lineColor: string;
  lineHeight?: number;
}
