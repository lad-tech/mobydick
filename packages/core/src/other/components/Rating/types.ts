import {StyleProp, ViewStyle} from 'react-native';

export type TStar = {
  filled: boolean;
  starId: number;
  setRating: (data: number) => void;
  currentSelected: number;
  iconSize: number;
  iconStyle?: StyleProp<ViewStyle>;
  fillColor?: string;
  disabled?: boolean;
};

export type TRating = {
  setCurrentRate?: (data: number) => void;
  count: number;
  iconStyle?: StyleProp<ViewStyle>;
  iconSize?: number;
  fillColor?: string;
  currentRate: number;
  disabled?: boolean;
};
