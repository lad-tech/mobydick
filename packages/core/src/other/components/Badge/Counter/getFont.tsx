import {TypographyProp} from '../../../../typography';

import {ICounterSize, ICounterTypes} from './types';

const getColorText = (type: ICounterTypes) => {
  switch (type) {
    case ICounterTypes.accentLight:
      return 'Accent';
    case ICounterTypes.attentionLight:
      return 'Error';
    case ICounterTypes.mutedLight:
      return 'Muted';
    default:
      return 'White';
  }
};
export const getFont = (
  type: ICounterTypes,
  size: ICounterSize,
): TypographyProp => {
  return `SemiBold-${getColorText(type)}-${size === ICounterSize.medium ? 'M' : 'XS'}`;
};
