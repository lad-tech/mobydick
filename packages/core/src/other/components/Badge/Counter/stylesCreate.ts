import {createStyles} from '../../../../styles';
import px from '../../../../styles/utils/px';

import {ICounterSize, ICounterTypes} from './types';

const stylesCreate = createStyles(
  ({spaces, colors}, size: ICounterSize, type?: ICounterTypes) => {
    const isMedium = size === ICounterSize.medium;
    const defaultSize = isMedium ? spaces.Space24 : px(18);

    const getBackgroundColor = () => {
      switch (type) {
        case ICounterTypes.attention:
          return {backgroundColor: colors.ElementAttention};
        case ICounterTypes.accent:
          return {backgroundColor: colors.ElementBase};
        case ICounterTypes.muted:
          return {backgroundColor: colors.ElementMuted};
        default:
          return {backgroundColor: colors.ElementWhite};
      }
    };

    return {
      counter: {
        zIndex: 1,
        alignSelf: 'center',
        minWidth: defaultSize,
        height: defaultSize,
        borderRadius: defaultSize / 2,
        alignItems: 'center',
        justifyContent: 'center',
        ...getBackgroundColor(),
      },
      text: {
        textAlign: 'center',
        paddingHorizontal: isMedium ? spaces.Space6 : spaces.Space4,
      },
    };
  },
);

export default stylesCreate;
