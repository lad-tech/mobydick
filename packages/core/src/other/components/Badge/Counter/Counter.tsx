import {FC} from 'react';

import View from '../../../../basic/components/View/View';
import useStyles from '../../../../styles/hooks/useStyles';
import {Typography} from '../../../../typography';

import stylesCreate from './stylesCreate';
import {ICounterProps, ICounterSize, ICounterTypes} from './types';
import {getFont} from './getFont';

const Counter: FC<ICounterProps> = ({
  count,
  style,
  textStyle,
  font,
  size = ICounterSize.medium,
  type = ICounterTypes.accent,
  maxLength = 2,
}) => {
  const [styles] = useStyles(stylesCreate, size, type);

  if (!count) {
    return null;
  }

  const lastNumber = `${'9'.repeat(maxLength)}+`;

  const text =
    count.toString().length > maxLength ? lastNumber : count.toString();

  return (
    <View style={[styles.counter, style]}>
      <Typography
        style={[styles.text, textStyle]}
        font={font ?? getFont(type, size)}>
        {text}
      </Typography>
    </View>
  );
};

export default Counter;
