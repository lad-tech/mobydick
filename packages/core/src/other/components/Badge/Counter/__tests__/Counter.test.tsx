import {render} from '@testing-library/react-native';

import Counter from '../Counter';
import {ICounterSize, ICounterTypes} from '../types';

describe('Counter', () => {
  test('render counter', () => {
    const {toJSON} = render(<Counter count={0} />);
    expect(toJSON()).toMatchSnapshot();
  });
  test('render counter ICounterSize.medium', () => {
    const {toJSON} = render(<Counter size={ICounterSize.medium} count={1} />);
    expect(toJSON()).toMatchSnapshot();
  });
  test('render counter ICounterSize.small', () => {
    const {toJSON} = render(<Counter size={ICounterSize.small} count={12} />);
    expect(toJSON()).toMatchSnapshot();
  });
  test('render counter ICounterTypes.accentLight', () => {
    const {toJSON} = render(
      <Counter type={ICounterTypes.accentLight} count={123} />,
    );
    expect(toJSON()).toMatchSnapshot();
  });
  test('render counter ICounterTypes.attentionLight', () => {
    const {toJSON} = render(
      <Counter type={ICounterTypes.attentionLight} count={2} />,
    );
    expect(toJSON()).toMatchSnapshot();
  });
  test('render counter ICounterTypes.attention', () => {
    const {toJSON} = render(
      <Counter type={ICounterTypes.attention} count={99} maxLength={1} />,
    );
    expect(toJSON()).toMatchSnapshot();
  });
  test('render counter ICounterTypes.accent', () => {
    const {toJSON} = render(<Counter type={ICounterTypes.accent} count={10} />);
    expect(toJSON()).toMatchSnapshot();
  });
  test('render counter ICounterTypes.mutedLight', () => {
    const {toJSON} = render(
      <Counter type={ICounterTypes.mutedLight} count={10} />,
    );
    expect(toJSON()).toMatchSnapshot();
  });
  test('render counter ICounterTypes.muted', () => {
    const {toJSON} = render(<Counter type={ICounterTypes.muted} count={10} />);
    expect(toJSON()).toMatchSnapshot();
  });
  test('render counter style', () => {
    const {toJSON} = render(
      <Counter style={{backgroundColor: '#FFF'}} count={20} />,
    );
    expect(toJSON()).toMatchSnapshot();
  });
  test('render notification', () => {
    const {toJSON} = render(<Counter />);
    expect(toJSON()).toMatchSnapshot();
  });
  test('render font', () => {
    const {toJSON} = render(<Counter font={'Bold-Black-XXS'} count={10} />);
    expect(toJSON()).toMatchSnapshot();
  });
});
