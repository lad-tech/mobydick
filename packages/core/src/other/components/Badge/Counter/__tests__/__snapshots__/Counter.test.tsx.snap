// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter render counter 1`] = `null`;

exports[`Counter render counter ICounterSize.medium 1`] = `
<View
  style={
    [
      {
        "alignItems": "center",
        "alignSelf": "center",
        "backgroundColor": "#2B78EE",
        "borderRadius": 12,
        "height": 24,
        "justifyContent": "center",
        "minWidth": 24,
        "zIndex": 1,
      },
      undefined,
    ]
  }
>
  <Text
    allowFontScaling={true}
    style={
      [
        {
          "color": "#FFFFFF",
          "fontFamily": "Inter",
          "fontSize": 16,
          "fontWeight": "600",
          "lineHeight": 20,
          "minHeight": 20,
        },
        [
          {
            "paddingHorizontal": 6,
            "textAlign": "center",
          },
          undefined,
        ],
      ]
    }
  >
    1
  </Text>
</View>
`;

exports[`Counter render counter ICounterSize.small 1`] = `
<View
  style={
    [
      {
        "alignItems": "center",
        "alignSelf": "center",
        "backgroundColor": "#2B78EE",
        "borderRadius": 9,
        "height": 18,
        "justifyContent": "center",
        "minWidth": 18,
        "zIndex": 1,
      },
      undefined,
    ]
  }
>
  <Text
    allowFontScaling={true}
    style={
      [
        {
          "color": "#FFFFFF",
          "fontFamily": "Inter",
          "fontSize": 10,
          "fontWeight": "600",
          "lineHeight": 14,
          "minHeight": 14,
        },
        [
          {
            "paddingHorizontal": 4,
            "textAlign": "center",
          },
          undefined,
        ],
      ]
    }
  >
    12
  </Text>
</View>
`;

exports[`Counter render counter ICounterTypes.accent 1`] = `
<View
  style={
    [
      {
        "alignItems": "center",
        "alignSelf": "center",
        "backgroundColor": "#2B78EE",
        "borderRadius": 12,
        "height": 24,
        "justifyContent": "center",
        "minWidth": 24,
        "zIndex": 1,
      },
      undefined,
    ]
  }
>
  <Text
    allowFontScaling={true}
    style={
      [
        {
          "color": "#FFFFFF",
          "fontFamily": "Inter",
          "fontSize": 16,
          "fontWeight": "600",
          "lineHeight": 20,
          "minHeight": 20,
        },
        [
          {
            "paddingHorizontal": 6,
            "textAlign": "center",
          },
          undefined,
        ],
      ]
    }
  >
    10
  </Text>
</View>
`;

exports[`Counter render counter ICounterTypes.accentLight 1`] = `
<View
  style={
    [
      {
        "alignItems": "center",
        "alignSelf": "center",
        "backgroundColor": "#FFFFFF",
        "borderRadius": 12,
        "height": 24,
        "justifyContent": "center",
        "minWidth": 24,
        "zIndex": 1,
      },
      undefined,
    ]
  }
>
  <Text
    allowFontScaling={true}
    style={
      [
        {
          "color": "#2B78EE",
          "fontFamily": "Inter",
          "fontSize": 16,
          "fontWeight": "600",
          "lineHeight": 20,
          "minHeight": 20,
        },
        [
          {
            "paddingHorizontal": 6,
            "textAlign": "center",
          },
          undefined,
        ],
      ]
    }
  >
    99+
  </Text>
</View>
`;

exports[`Counter render counter ICounterTypes.attention 1`] = `
<View
  style={
    [
      {
        "alignItems": "center",
        "alignSelf": "center",
        "backgroundColor": "#F54D3D",
        "borderRadius": 12,
        "height": 24,
        "justifyContent": "center",
        "minWidth": 24,
        "zIndex": 1,
      },
      undefined,
    ]
  }
>
  <Text
    allowFontScaling={true}
    style={
      [
        {
          "color": "#FFFFFF",
          "fontFamily": "Inter",
          "fontSize": 16,
          "fontWeight": "600",
          "lineHeight": 20,
          "minHeight": 20,
        },
        [
          {
            "paddingHorizontal": 6,
            "textAlign": "center",
          },
          undefined,
        ],
      ]
    }
  >
    9+
  </Text>
</View>
`;

exports[`Counter render counter ICounterTypes.attentionLight 1`] = `
<View
  style={
    [
      {
        "alignItems": "center",
        "alignSelf": "center",
        "backgroundColor": "#FFFFFF",
        "borderRadius": 12,
        "height": 24,
        "justifyContent": "center",
        "minWidth": 24,
        "zIndex": 1,
      },
      undefined,
    ]
  }
>
  <Text
    allowFontScaling={true}
    style={
      [
        {
          "color": "#F54C3D",
          "fontFamily": "Inter",
          "fontSize": 16,
          "fontWeight": "600",
          "lineHeight": 20,
          "minHeight": 20,
        },
        [
          {
            "paddingHorizontal": 6,
            "textAlign": "center",
          },
          undefined,
        ],
      ]
    }
  >
    2
  </Text>
</View>
`;

exports[`Counter render counter ICounterTypes.muted 1`] = `
<View
  style={
    [
      {
        "alignItems": "center",
        "alignSelf": "center",
        "backgroundColor": "#D0D6E0",
        "borderRadius": 12,
        "height": 24,
        "justifyContent": "center",
        "minWidth": 24,
        "zIndex": 1,
      },
      undefined,
    ]
  }
>
  <Text
    allowFontScaling={true}
    style={
      [
        {
          "color": "#FFFFFF",
          "fontFamily": "Inter",
          "fontSize": 16,
          "fontWeight": "600",
          "lineHeight": 20,
          "minHeight": 20,
        },
        [
          {
            "paddingHorizontal": 6,
            "textAlign": "center",
          },
          undefined,
        ],
      ]
    }
  >
    10
  </Text>
</View>
`;

exports[`Counter render counter ICounterTypes.mutedLight 1`] = `
<View
  style={
    [
      {
        "alignItems": "center",
        "alignSelf": "center",
        "backgroundColor": "#FFFFFF",
        "borderRadius": 12,
        "height": 24,
        "justifyContent": "center",
        "minWidth": 24,
        "zIndex": 1,
      },
      undefined,
    ]
  }
>
  <Text
    allowFontScaling={true}
    style={
      [
        {
          "color": "#9BA1B0",
          "fontFamily": "Inter",
          "fontSize": 16,
          "fontWeight": "600",
          "lineHeight": 20,
          "minHeight": 20,
        },
        [
          {
            "paddingHorizontal": 6,
            "textAlign": "center",
          },
          undefined,
        ],
      ]
    }
  >
    10
  </Text>
</View>
`;

exports[`Counter render counter style 1`] = `
<View
  style={
    [
      {
        "alignItems": "center",
        "alignSelf": "center",
        "backgroundColor": "#2B78EE",
        "borderRadius": 12,
        "height": 24,
        "justifyContent": "center",
        "minWidth": 24,
        "zIndex": 1,
      },
      {
        "backgroundColor": "#FFF",
      },
    ]
  }
>
  <Text
    allowFontScaling={true}
    style={
      [
        {
          "color": "#FFFFFF",
          "fontFamily": "Inter",
          "fontSize": 16,
          "fontWeight": "600",
          "lineHeight": 20,
          "minHeight": 20,
        },
        [
          {
            "paddingHorizontal": 6,
            "textAlign": "center",
          },
          undefined,
        ],
      ]
    }
  >
    20
  </Text>
</View>
`;

exports[`Counter render font 1`] = `
<View
  style={
    [
      {
        "alignItems": "center",
        "alignSelf": "center",
        "backgroundColor": "#2B78EE",
        "borderRadius": 12,
        "height": 24,
        "justifyContent": "center",
        "minWidth": 24,
        "zIndex": 1,
      },
      undefined,
    ]
  }
>
  <Text
    allowFontScaling={true}
    style={
      [
        {
          "color": "#151C2C",
          "fontFamily": "Inter",
          "fontSize": 10,
          "fontWeight": "700",
          "lineHeight": 14,
          "minHeight": 14,
        },
        [
          {
            "paddingHorizontal": 6,
            "textAlign": "center",
          },
          undefined,
        ],
      ]
    }
  >
    10
  </Text>
</View>
`;

exports[`Counter render notification 1`] = `null`;
